src/slickml/utils/_transform.py

```diff
@@ -0,0 +1,88 @@
+from typing import Optional
+
+import pandas as pd
+from scipy.sparse import csr_matrix
+
+
+def memory_use_csr(csr: csr_matrix) -> int:
+    """Memory use of a Compressed Sparse Row (CSR) matrix in bytes.
+
+    Parameters
+    ----------
+    csr : csr_matrix
+        Compressed sparse row matrix
+
+    Returns
+    -------
+    int
+        Memory use in bytes
+    """
+    if not isinstance(csr, csr_matrix):
+        raise TypeError("The input csr must have scipy.sparse.csr_matrix dtype.")
+
+    return csr.data.nbytes + csr.indptr.nbytes + csr.indices.nbytes
+
+
+# TODO(amir): currently `pydantic.validate_arguments` is in beta version and they dont supoort
+# pandas.core: https://pydantic-docs.helpmanual.io/usage/validation_decorator/
+# we eventually should be able to use it to decorate our functions and remove the validations pieces
+def df_to_csr(
+    df: pd.DataFrame,
+    *,
```

Comment:
We are gonna ENFORE this from now on; to make sure named args rather positional args.